Add custom keystore documentations (#5297)
wso2/docs-is: en/includes/deploy/security/keystores/configure-custom-keystores.md

# Configure Custom KeyStores for Authentication Protocols

> By default, WSO2 Identity Server uses the private key and certificate from the Primary KeyStore in all authentication protocols such as,

Reviewer: ```suggestion
By default, WSO2 Identity Server uses the private key and certificate from the primary keystore in all authentication protocols such as,
```

Author: addressed by https://github.com/wso2/docs-is/pull/5297/commits/150dc2057392db4922a85d4973fcba77dab36f59

> - Sign JWT tokens in OAuth authentication flows  
> - Verify JWT tokens signed by WSO2 identity server  
> - Sign SAML assertions in SAML authentication flows

Reviewer: ```suggestion
- Signing JWT tokens in OAuth authentication flows  
- Verifying JWT tokens signed by WSO2 identity server  
- Signing SAML assertions in SAML authentication flows
```

Author: addressed by https://github.com/wso2/docs-is/pull/5297/commits/150dc2057392db4922a85d4973fcba77dab36f59

WSO2 Identity Server also allows configuring separate keystores for each authentication protocol.

> !!! note

Reviewer: shall we move this note below the "this document provides.." phrase?

Author: addressed by https://github.com/wso2/docs-is/pull/5297/commits/150dc2057392db4922a85d4973fcba77dab36f59

    Currently only the following authentication protocols are supported,

    - OAuth 2.0
    - WS-Federation
    - WS-Trust (Needs to be added via the WS-Trust connector)
    - SAML

> This document provides step-by-step instructions for configuring a custom keystore for a specific authentication protocol. You can define multiple custom keystores and assign a separate keystore to each protocol for better management and security.

Reviewer: This document provides step-by-step instructions to configure a custom keystore for a specific authentication protocol. You can define multiple custom keystores and assign each to a different protocol as needed.

Author: addressed by https://github.com/wso2/docs-is/pull/5297/commits/150dc2057392db4922a85d4973fcba77dab36f59

## Configure Custom KeyStore for OAuth, WS-Trust or WS-Federation

> 1. Place your custom key store file in `<IS-HOME>/repository/resources/security/` directory.  

Reviewer: ```suggestion
1. Locate your custom key store file in `<IS-HOME>/repository/resources/security/` directory.  
```

Author: addressed by https://github.com/wso2/docs-is/pull/5297/commits/150dc2057392db4922a85d4973fcba77dab36f59

2. Add the following configuration to `deployment.toml` file.

    === "JKS"
        ``` toml
        [[keystore.custom]]
        file_name = "<keystore name>"
        password = "<password>"
        type = "JKS"
        alias = "<private key alias>"
        key_password = "<password>"
        ```

    === "PKCS12"
        ``` toml
        [[keystore.custom]]
        file_name = "<keystore name>"
        password = "<password>"
        type = "PKCS12"
        alias = "<private key alias>"
        key_password = "<password>"
        ```

    ### Configuration Parameters,

    | Configuration | Details |
    | :---- | :---- |
    | `file_name` | Key store file name. |
    | `password` | Key Store password. |
    | `type` | Key Store Type. |
    | `alias` | Alias value of the private key. |

>     | `key_password` | Private key password. |

Reviewer: shall we align this to the left most side?

Author: This couldn't be done because the section is defined under a numbered point, and making this change would alter the layout.

Author: I removed the heading and added it as regular text because a heading is not appropriate in this context.

Reviewer: Ack.

3. Map the Custom KeyStore to Authentication Protocols  

    - This configuration maps the custom key stores configured above to different authentication protocols.  

>     - As of now, the following authentication protocols are supported,

Reviewer: Shall we make this a description instead of list of points?

Author: addressed by https://github.com/wso2/docs-is/pull/5297/commits/150dc2057392db4922a85d4973fcba77dab36f59
